models/issues/stopwatch.go

```diff
@@ -133,10 +135,24 @@ func StopwatchExists(userID, issueID int64) bool {
 
 // HasUserStopwatch returns true if the user has a stopwatch
 func HasUserStopwatch(ctx context.Context, userID int64) (exists bool, sw *Stopwatch, err error) {
-	sw = new(Stopwatch)
+	type stopwatchIssueRepo struct {
+		Stopwatch       `xorm:"extends"`
+		Issue           `xorm:"extends"`
+		repo.Repository `xorm:"extends"`
+	}
```

Comment:
I am not sure whether this is best practice.

The conflicted names will overwrite unrelated model's fields?

IIRC, in such case, `Stopwatch.id == Issue.id == Repoository.id` ?

Reply:
The SQL returns stopwatch.id, issue.id and repository.id it does not return id, id, id. Xorm uses the full name to match against the struct not the short name. There are no conflicted names.

This use of extends is found elsewhere in gitea code and works correctly. It is described in the xorm documentation and iirc there is a test for it with in the xorm code base.

I'm not sure what makes you think it's not best practice.

Reply:
I am not sure because I didn't know how XORM works. Thank you for the explanation, then it is the best practice.